orchestrator/relayer/src/fee_manager.rs

```diff
@@ -127,29 +91,81 @@ impl FeeManager {
         batch_fee: &Erc20Token,
         contract_address: &EthAddress,
     ) -> bool {
-        if self.mode == RelayerMode::AlwaysRelay {
-            return true;
+        match self.mode {
+            RelayerMode::AlwaysRelay => {
+                return true;
+            }
+            RelayerMode::File => {
+                if self.should_send_at_non_profitable_cost(contract_address) {
+                    return true
+                }
+                let token_price = match self.get_token_price(&batch_fee.token_contract_address).await {
+                    Ok(token_price) => token_price,
+                    Err(_) => return false,
+                };
+
+                let estimated_fee = estimated_cost.get_total();
+                let batch_value = batch_fee.amount * token_price;
+
+                info!("estimate cost is {}, batch value is {}", estimated_fee, batch_value);
+                return batch_value >= estimated_fee
+            }
+            RelayerMode::Api => {
+                let body = json!({
+                    "batchFee": {
+                        "amount": batch_fee.amount.as_u64(),
+                        "tokenContractAddress": batch_fee.token_contract_address
+                    },
+                    "estimatedCost": {
+                        "gas": estimated_cost.gas.as_u64(),
+                        "gasPrice": estimated_cost.gas_price.as_u64()
+                    }}
+                );
+
+                return match Client::new()
+                    .post(self.relayer_api_url.as_str())
+                    .json(&body)
+                    .send().await {
+                    Ok(resp) => match resp.json().await {
+                        Ok(json) => {
+                            let api_response: ApiResponse = json;
+                            if api_response.in_blacklist {
+                                return false
+                            }
+                            {
+                                api_response.profitable ||
+                                    self.should_send_at_non_profitable_cost(contract_address)
+                            }
+                        }
+                        Err(err) => {
+                            error!("error on relayer api: {}", err);
+                            false
+                        }
+                    }
+                    Err(err) => {
+                        error!("error on relayer api: {}", err);
+                        false
+                    }
+                }
+            }
         }
+    }
 
+    fn should_send_at_non_profitable_cost(
+        &mut self,
+        contract_address: &EthAddress,
+    ) -> bool {
         match self.next_batch_send_time.get(contract_address) {
             Some(time) => {
                 if *time < Instant::now() {
                     return true;
+                } else {
+                    return false;
```

Comment:
it can just be `return *time < Instant::now();`?

Reply:
yes, fixed